design/variable_command_design.md

```diff
@@ -0,0 +1,181 @@
+# Dockerless CLI - variable
+
+This document is the design document for the variable CLI command for the Dockerless project. The functionality for the variable CLI command already exists so all that is left is the following:
+
+1. Ensure UTs and integration tests exist and are GA quality
+2. Improve error handling experience
+3. Write proper documentation in `README.md` of the repository
+
+
+
+```bash
+Name
+----
+  variable - Manage variables
+
+
+Usage
+----
+  conjur variable get [arguments...]
+
+  conjur variable set [arguments...]
+
+
+Global options
+----
+  -h, --help show this help message and exit
+
+
+Commands
+----
+  get       Get the value of a variable can be applied to get the value of several variables
+
+  set       Set the value of a variable
+
+To get help on a specific command, type: conjur <command> -h
+```
+
+
+
+### UX mapping
+
+The following is a mapping of the current flow for all possible permutations of the variable CLI command and their respective expected outputs.
+
+*Input for variable is neither get/set*
+
+```bash
+$ conjur-cli variable boo
+
+## Current
+usage: conjur-cli variable [-h] {get,set} ...
+conjur-cli variable: error: argument action: invalid choice: 'boo' (choose from 'get', 'set')
+
+## Should be
+Error: Command 'variable' requires a subcommand get, set
+
+<Variable command help screen>
+```
+
+*One of the batch secrets doesn’t have value/doesn’t exist*
+
+```bash
+$ conjur-cli variable get secrets/test_secret foo/bar
+
+## Current
+requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://localhost/secrets/myorg/variable/secrets%2Ftest_secret%2Cfoot%2Fbar
+
+## Should be
+Error: Data either not found or variable value does not exist
+
+<Variable command help screen>
+```
+
+*Set secrets*
+
+```bash
+$ conjur-cli variable set foo/bar mysecret
+
+## Current
+Value set: 'foo/bar'
+
+## Should be
+Success! Data written to: 'foo/bar'
```

Comment:
Out of curiousity, what makes this non-machine readable? The `''` and the `!`?

Reply:
If you try to give me a way to parse the variable name from each of those outputs, you will see that the suggested use case requires careful counting vs otherwise. Both can be extracted the same way but the original output is less prone to error during script development.

Reply:
why careful counting if we split at the spaces and take the last index. Can you explain a bit more about what you mean by this or why current output makes it less error prone?

Reply:
As we spoke about that the writing variables isn’t critical as is reading of variable be machine-readable so it’s ok